src/redux/features/appVersion/appVersionSlice.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+// Define a type for the slice state
+interface AppVersionState {
+  version?: string;
+}
+
+// Define the initial state using that type
+const initialState: AppVersionState = {};
+
+export const appVersionSlice = createSlice({
+  name: 'appVersion',
+  initialState,
+  reducers: {
+    setVersionAction: (state, action: PayloadAction<string>) => {
+      state.version = action.payload;
+    },
+  },
+});
+
+export const { setVersionAction } = appVersionSlice.actions;
```

Comment:
notice the `xAction` naming convention. This is helpful for reminding implementers that this function needs to be passed through a `dispatch()` in order to work